generators/app/templates/src/screens/Login/index.tsx

```diff
@@ -1,50 +1,98 @@
-import React, { useState } from 'react';
+import React, { useCallback } from 'react';
 import { useHistory } from 'react-router-dom';
+import i18next from 'i18next';
+import { useForm } from 'react-hook-form';
 
+import { useLazyRequest } from 'hooks/useRequest';
 import { useDispatch } from 'contexts/UserContext';
 import { actionCreators, Credentials, User } from 'contexts/UserContext/reducer';
-import { useLazyRequest } from 'hooks/useRequest';
 import { login, setCurrentUser } from 'services/AuthServices';
+import FormInput from 'components/FormInput';
+import PATHS from 'components/Routes/paths';
+
+import styles from './styles.module.scss';
 
-import Login from './layout';
+const FIELDS = { email: 'email', password: 'password' };
+
+const VALIDATIONS = {
+  email: {
+    pattern: {
+      value: /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+      message: i18next.t('Login:emailFormatError') as string
+    },
+    required: i18next.t('Login:requiredError') as string
+  },
+  password: { required: i18next.t('Login:requiredError') as string }
+};
 
 function LoginContainer() {
   const history = useHistory();
   const dispatch = useDispatch();
-  const [email, setEmail] = useState('');
-  const [password, setPassword] = useState('');
+
   const [, loading, loginError, loginRequest] = useLazyRequest({
     request: (credentials: Credentials) => login(credentials),
     withPostSuccess: response => {
       const userResponse = response as User;
       dispatch(actionCreators.setUser(userResponse));
       setCurrentUser(userResponse);
+
       history.push('/');
     }
   });
+  const { register, handleSubmit, errors } = useForm();
+  const errorMessage = loginError?.errorData?.message;
 
-  const handleLogin = (event: React.FormEvent<HTMLFormElement>) => {
-    event.preventDefault();
-    const credentials = { username: email, password };
-    dispatch(actionCreators.login(credentials));
-    loginRequest(credentials);
-  };
-
-  const handleEmailChange = (e: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>) =>
-    setEmail((e.target as HTMLInputElement).value || '');
-
-  const handlePasswordChange = (e: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>) => {
-    setPassword((e.target as HTMLInputElement).value || '');
-  };
+  const handleLogin = useCallback(
```

Comment:
if you type `useForm<Credentials>` you dont have to type submit handler, nor errors